src/tendril.rs

```diff
@@ -85,30 +179,40 @@ pub enum SubtendrilError {
 /// with one of the marker types from `tendril::fmt`. See the `StrTendril`
 /// and `ByteTendril` type aliases for two examples.
 ///
+/// The type parameter `A` indicates the atomicity of the tendril; it is by
+/// default `NonAtomic`, but can be specified as `Atomic` to get a tendril
+/// which implements `Send` and `Sync` (viz. a thread-safe tendril).
+///
 /// The maximum length of a `Tendril` is 4 GB. The library will panic if
 /// you attempt to go over the limit.
 #[cfg_attr(feature = "unstable", unsafe_no_drop_flag)]
 #[repr(packed)]
-pub struct Tendril<F>
+pub struct Tendril<F, A = NonAtomic>
     where F: fmt::Format,
+          A: Atomicity,
 {
     ptr: Cell<NonZero<usize>>,
     len: u32,
     aux: Cell<u32>,
     marker: PhantomData<*mut F>,
+    refcount_marker: PhantomData<A>,
 }
 
+unsafe impl<F, A> Send for Tendril<F, A> where F: fmt::Format, A: Atomicity + Sync { }
+unsafe impl<F, A> Sync for Tendril<F, A> where F: fmt::Format, A: Atomicity + Sync { }
```

Comment:
@chris-morgan Reading this again, I think the `Send` impl here is safe but I’m not convinced about the `Sync` one, since `Tendril` contains `Cell`s. Can you explain the reasoning of why this is safe?